draft-yasskin-http-origin-signed-responses.md

```diff
@@ -1039,14 +1037,27 @@ an exchange with an HTTPS request URI provides a TLS guarantee that the exchange
 isn't out of date (as long as {{oq-cross-origin-push}} is resolved to keep the
 same-origin requirement).
 
-## Signing oracles are permanent ## {#seccons-signing-oracles}
+## Signing oracles are defended by OCSP ## {#seccons-signing-oracles}
+
+There are several reasons a signing oracle for a private key may be accidentally
+exposed without exposing the private key itself. For example, organizations that
+run edge caches may provide a signing oracle to those machines. Those machines
+may be less physically secure than the machines with actual access to the TLS
+private key. When an edge cache is compromised, this allows recovery process to
+be as simple as turning off its signing oracle and waiting for clients to close
+compromised connections, rather than revoking the whole private key.
+
+Thus, signed exchanges cannot allow access to a signing oracle to allow minting
+exchanges that are valid long after the signing oracle is closed. To prevent
+this, signatures are required to cover the current OCSP response for a
+signature. This naturally limits their validity to the life of the latest OCSP
+response available when the signing oracle was closed. To limit this to 7 days,
+CAs SHOULD NOT pre-sign future OCSP responses.
 
-An attacker with temporary access to a signing oracle can sign "still valid"
-assertions with arbitrary timestamps and expiration times. As a result, when a
-signing oracle is removed, the keys it provided access to SHOULD be revoked so
-that, even if the attacker used them to sign future-dated exchange validity
-assertions, the key's OCSP assertion will expire, causing the exchange as a
-whole to become untrusted.
+Another approach, that this specification doesn't take yet, is to avoid re-using
```

Comment:
Not sure what "yet" means in this context. Going from opt-in X.509 extension to this weaker OCSP trick is a meaningful direction, but we can't switch from this weaker option to an X.509 extension because the mere existence of the other option is what introduces the vulnerability.

(Or did you mean this as more a TODO-like comment rather than final specification text?)

Reply:
It's something like a TODO, yeah. I've changed it.